chimney/src/main/scala/io/scalaland/chimney/ValueProvider.scala

```diff
@@ -0,0 +1,35 @@
+package io.scalaland.chimney
+
+import shapeless.{::, HList, HNil, LabelledGeneric, ops}
+
+trait ValueProvider[FromG, From, T, Label <: Symbol, Modifiers <: HList] {
+  def provide(from: FromG, modifiers: Modifiers): T
+}
+
+object ValueProvider {
+
+  implicit def hnilCase[FromG <: HList, L <: Symbol, T1, T, From]
+  (implicit fieldSelector: ops.record.Selector.Aux[FromG, L, T1],
+   fieldTransformer: DerivedTransformer[T1, T, HNil])
```

Comment:
Similar to above - some types has more descriptive name, some are just `T` and `T1`.  Also we could keep order of parameters between implicit defs. E.g.

    hnilCase[FromG <: HList, From, L <: Symbol. FromT, ToT]

(just a suggestion).